cli/azd/pkg/oneauth/c_funcs.go

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//go:build oneauth
+
+package oneauth
+
+/*
+extern void goLog(char *s);
+
+// enables native code to call goLog (can't pass a Go function pointer to C)
```

Comment:
Learned about https://pkg.go.dev/syscall?GOOS=windows#NewCallback yesterday, which includes this terrifying comment:

> Only a limited number of callbacks may be created in a single Go process, and any memory allocated for these callbacks is never released. Between NewCallback and NewCallbackCDecl, at least 1024 callbacks can always be created.

Thought this would make you smile, so sharing it here - what you have is obviously better given that we are already in the CGo world anyway...